api/ocm/extensions/repositories/genericocireg/componentversion.go

```diff
@@ -304,15 +321,51 @@ func (c *ComponentVersionContainer) AddBlob(blob cpi.BlobAccess, refName string,
 		return nil, errors.New("a resource has to be defined")
 	}
 
+	size := blob.Size()
+	limit := c.comp.repo.blobLimit
+	var refs []string
+	if limit > 0 && size != blobaccess.BLOB_UNKNOWN_SIZE && size > limit {
+		reader, err := blob.Reader()
+		if err != nil {
+			return nil, err
+		}
+		defer reader.Close()
+		ch := chunked.New(reader, limit, vfsattr.Get(c.GetContext()))
+		for {
+			b, err := ch.Next()
+			if err != nil {
+				return nil, errors.Wrapf(err, "chunked blob")
+			}
+			if b == nil {
+				break
+			}
+			err = c.addLayer(b, &refs)
+			b.Close()
+
+			if err != nil {
+				return nil, errors.Wrapf(err, "chunked blob")
+			}
+		}
```

Comment:
couldnt we replace that with something like this
```
		for {
			br, err := func() (bool, error) {
				f, err := blobaccess.NewTempFile("", "chunk-*", vfsattr.Get(c.GetContext()))
				if err != nil {
					return true, err
				}
				defer f.Close()
				written, err := io.CopyN(f.Writer(), reader, limit)
				if err != nil && err != io.EOF {
					if err := f.Close(); err != nil {
						errors.Join(err, errors.Wrapf(err, "close temp file"))
					}
					return true, err
				}
				if written != 0 {
					b := f.AsBlob(blob.MimeType())
					err = c.addLayer(b, &refs)
					if err != nil {
						return true, err
					}
				}
				if err == io.EOF {
					return true, nil
				}
				return false, nil
			}()
			if err != nil {
				return nil, err
			}
			if br {
				break
			}
		}
```
and get rid of the `ChunkedReader`, the `LookAheadReader` and the `ChunkedBlobAccess`?

Reply:
sure, could also do something like this

```
BlobAccessForChunk := func(r io.Reader, chunk int64) (cpi.BlobAccess, bool, error) {
	f, err := blobaccess.NewTempFile("", "chunk-*", vfsattr.Get(c.GetContext()))
	if err != nil {
		return nil, true, err
	}
	defer f.Close()
	written, err := io.CopyN(f.Writer(), reader, limit)
	if err != nil && err != io.EOF {
		return nil, true, err
	}
	if written <= 0 {
		return nil, true, fmt.Errorf("no data written")
	}
	return f.AsBlob(blob.MimeType()), err == io.EOF, nil
}

for {
	b, br, err := BlobAccessForChunk(reader, limit)
	if err != nil {
		return nil, err
	}
	defer b.Close()

	err = c.addLayer(b, &refs)
	if err != nil {
		return nil, err
	}
	if br {
		break
	}
}
```